components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -1175,4 +1175,22 @@ public static void addDefaultAuthProviders(List<String> authProviders) {
         authProviders.add(OpenAPIConstants.APISecurity.oauth2.name());
         authProviders.add(OpenAPIConstants.APISecurity.jwt.name());
     }
+
+    private static String resolveBasePathFromContextTemplate(Map<String, Object> extensions, String version) {
+        String basePath = extensions.get(OpenAPIConstants.BASEPATH).toString();
```

Comment:
Doesn't this cause NPE if basepath extension is not defined?

Reply:
There is basepath validation method which checks and set  the base path to the extension. So at this point extension is always available